packages/altair-app/src/app/services/subscriptions/providers/app-sync.ts

```diff
@@ -23,14 +24,13 @@ export class AppSyncSubscriptionProvider extends SubscriptionProvider {
     const region = this.connectionParams.aws_appsync_region;
     const auth = {
       type: this.connectionParams.aws_appsync_authenticationType,
-      apiKey: this.connectionParams.aws_appsync_apiKey,
+      ...this.connectionParams.aws_appsync_apiKey ? {apiKey: this.connectionParams.aws_appsync_apiKey} : {},
```

Comment:
Is there any need to conditionally add it? If the user doesn't provide it, it should be undefined which should be the same thing.

Reply:
@imolorhe , I am not sure about that as I only have one AppSync server with me. Based on the AWS document, it is just either one, so that's why I add the condition in it

Reply:
Okay but the condition doesn't choose one over the other. If the user specifies both, then both would still be added.

Reply:
Perhaps a better way to manage this would be to set one of them based on the value of `aws_appsync_authenticationType`

Reply:
Yes, that is true. I just pass the attribute whatever the user input.
The other possible is , we make aws_appsync_apiKey and jwtToken into one field, and based on the authentication type to add the parameter

Reply:
@imolorhe, we have the same idea at the same time. What is the new field name do you want to call? (For the apiKey and the jwtToken)

if type = API_KEY , then set to apiKey, otherwise jwtToken, do you think it is good enough? (or you want that will only be set if it is OPEN ID or AWS Cognito)

Reply:
If the user specifies API_KEY, then set the api key, else set the jwt token. There appears to be even more options for authentication than we have here, so this might be a bit more difficult.
https://github.com/aws-amplify/amplify-js/blob/e1d122bec0/packages/api-graphql/src/GraphQLAPI.ts#L125-L171

For now though, we can do it that way.

Reply:
@imolorhe , for the IAM, it is impossible to test with the Alatir, unless we support to load the AWS profile from CLI or Cross-Original setup.
OPEN ID and AWS Cognito User Pool are using JWT Token, so there are no different
API Key is the API Key

But yes, if AWS AppSync supports the new method in the future than we need to have a better logic

Reply:
I will do a quick test to see if the API_KEY = undefined or other value if it causes any issue for the OPEN ID Connect

Reply:
@imolorhe, i have updated the code, the conditional check is removed. We can assign as is. I have tested if it is undefined or with value, it will not break the Open ID Connector connection

Reply:
Awesome. Thanks!